### PR TITLE
GitHub Authentication

### PR DESCRIPTION
**Pull request by lightnet328**
## Overview
I implemented a feature to log in to Crowi using GitHub account.

## Features
- Log in to Crowi using GitHub account
- Login restrictions by GitHub Organization

## Changed
- Callback url when registering with a Google Account
  - `before: /register`
  - `after: /register/google`
  - It just behaves like a login callback

### CODE REVIEW

**Comment by sotarok** [comment]
Thank you for the great PR.
I'll check it.

File: lib/routes/login.js [hunk 1]
```diff
@@ -258,55 +323,107 @@ module.exports = function(crowi, app) {
...
+      var socialName = req.session.googleName || req.session.githubName;
+      var socialImage = req.session.googleImage || req.session.githubImage;
+      var issuerName = googleId ? "Google" : githubId ? "GitHub" : "";
+
+      if (!User.isEmailValid(socialEmail)) {
+        req.flash('registerWarningMessage', 'このメールアドレスのアカウントはコネクトできません。');
+        return res.redirect('/login?register=1');
+      }
+      if (githubOrganizations && !User.isGitHubAccountValid(githubOrganizations)) {
+        req.flash('registerWarningMessage', 'このアカウントはコネクトできません。');
```
Comment (sotarok) [inline lib/routes/login.js]: Can you make above of two messages English?

**Comment by sotarok** [comment]
@lightnet328 Can you merge current master to resolve conflicts?

File: lib/routes/me.js [hunk 2]
```diff
@@ -267,5 +269,67 @@ module.exports = function(crowi, app) {
     });
   };
 
+  actions.authGitHub = function(req, res, next) {
+    var githubAuth = require('../util/githubAuth')(config);
+
+    var userData = req.user;
+
+    var toDisconnect = req.body.disconnectGitHub ? true : false;
```
Comment (hiroppy) [inline lib/routes/me.js]:
```js
const toDisconnect = !!req.body.disconnectGitHub;
```

File: lib/views/me/index.html [hunk 3]
```diff
@@ -227,7 +225,66 @@ <h1 class="title" id="">{{ t('User Settings') }}</h1>
...
+                {{ t('page_me.form_help.github_disconnect1') }}<br>
+                {{ t('page_me.form_help.github_disconnect2') }}
+              </p>
+            </div>
+
+            {% else %}
+
+            <div class="col-sm-12">
+              <div class="text-center">
+                <input type="submit" name="connectGitHub" class="btn btn-github" value="GitHubコネクト">
```
[the remark this answers is not in the document]
Comment (lightnet328) [reply lib/views/me/index.html]: Well, it seems that it was not in a state to work after refactoring...
I fixed and translated it now.

**Pull request merged**